### PR TITLE
Add `chainerx.log2`

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
Ref. #6423

### CODE REVIEW

**Comment by keisuke-umezawa** [comment]
@asi1024 
Thank you for reviewing!
BTW, do you have any materials to understand the archtecture of chinerx? I want to understand whole archtecture of chainerx.

**Comment by asi1024** [comment]
Sorry, but we have made no materials other than #6423 yet.
We are aware that there is a demand for it and are currently planning to create one.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

File: chainerx_cc/chainerx/cuda/float16.cuh [hunk 1]
```diff
@@ -71,6 +71,7 @@ public:
     __device__ Float16 Exp() const { return Float16{std::exp(static_cast<float>(*this))}; }
     __device__ Float16 Log() const { return Float16{std::log(static_cast<float>(*this))}; }
     __device__ Float16 Log10() const { return Float16{std::log10(static_cast<float>(*this))}; }
+    __device__ Float16 Log2() const { return Float16{std::log2(static_cast<float>(*this))}; }
```
Comment (asi1024) [inline chainerx_cc/chainerx/cuda/float16.cuh]: `std::log2` seems not supported in CUDA. Could you fix it likes as follows?
```suggestion
    __device__ Float16 Log2() const { return Float16{std::log(static_cast<float>(*this)) / std::log(2.0) }; }
```

File: chainerx_cc/chainerx/cuda/numeric.cuh [hunk 2]
```diff
@@ -74,6 +74,7 @@ CHAINERX_DEFINE_CUDA_FLOAT16_FALLBACK_UNARY(Arctan, std::atan)
 CHAINERX_DEFINE_CUDA_FLOAT16_FALLBACK_UNARY(Exp, std::exp)
 CHAINERX_DEFINE_CUDA_FLOAT16_FALLBACK_UNARY(Log, std::log)
 CHAINERX_DEFINE_CUDA_FLOAT16_FALLBACK_UNARY(Log10, std::log10)
+CHAINERX_DEFINE_CUDA_FLOAT16_FALLBACK_UNARY(Log2, std::log2)
```
Comment (asi1024) [inline chainerx_cc/chainerx/cuda/numeric.cuh]: Ditto.
```cpp
__device__ inline double Log2(double x) { return std::log(x) / std::log(2.0); }
__device__ inline float Log2(float x) { return std::log(x) / std::log(2.0f); }
__device__ inline cuda::Float16 Log2(cuda::Float16 x) { return cuda::Float16{std::log(static_cast<float>(x)) / std::log(2.0f)}; }
```

**Comment by keisuke-umezawa** [comment]
@asi1024 I fixed them.

File: chainerx_cc/chainerx/cuda/float16.cuh (hunk 1 above)
Comment (asi1024) [reply chainerx_cc/chainerx/cuda/float16.cuh]: Sorry, the above my suggestion seems wrong. Could you fix as follows? 🙇 
```cpp
__device__ inline double Log2(double x) { return std::log2(x); }
__device__ inline float Log2(float x) { return std::log2f(x); }
__device__ inline cuda::Float16 Log2(cuda::Float16 x) { return cuda::Float16{std::log2f(static_cast<float>(x))}; }
```

Comment (keisuke-umezawa) [reply chainerx_cc/chainerx/cuda/float16.cuh]: Isn't it for `numeric.cuh`?

Comment (keisuke-umezawa) [reply chainerx_cc/chainerx/cuda/float16.cuh]: For here,

```
__device__ Float16 Log2() const { return Float16{std::log2f(static_cast<float>(*this))}; }
```

**Comment by asi1024** [comment]
Could you resolve conflicts?

**Comment by keisuke-umezawa** [comment]
@asi1024 Resolved conflicts.

File: chainerx_cc/chainerx/routines/explog.cc [hunk 3]
```diff
@@ -123,6 +123,27 @@ Array Log10(const Array& x) {
     return out;
 }
 
+Array Log2(const Array& x) {
```
Comment (asi1024) [inline chainerx_cc/chainerx/routines/explog.cc]: Could you add a declaration in `routines/explog.h`?

Comment (keisuke-umezawa) [reply chainerx_cc/chainerx/routines/explog.cc]: Sorry. I added.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM! Thank you for the PR!

**Pull request merged**